doc/ADDB_Monitoring.rst

```diff
@@ -0,0 +1,281 @@
+
+ADDB monitors serve two purposes:
+
+- To support online statistics reporting, like similar to df, vmstat, top. This is needed both globally (summarized over all nodes) and locally. 
+
+- Inform external world (HA & cluster management tools) about exceptional conditions like failures, overload, etc.
+
+***************
+ Definitions
+***************
+
+ADDB Analysis and Diagnostic Data Base. See [0] for additional details. 
+
+ADDB base record type [Text Wrapping Break]The architecture defines “event”, “data point” and “counter” type ADDB record categories.  See [0] for additional details.  In this HLD we will use the word “exception” instead of “event”.  We’ll use the term “event” to describe any derivation of the base record types. 
+
+Summary ADDB records 
+
+- These are summary records that are generated by ADDB monitor for a particular system metric. 
+
+ADDB monitors
+
+- These are objects present on every node in the cluster (client or server), they generate the summary records. 
+
+***************
+ Requirements
+*************** 
+
+- [r.addb.monitor.add.runtime] Addition of ADDB monitors. 
+
+- [r.addb.monitor.remove.runtime] Deletion of ADDB monitors. 
+
+- [r.addb.monitor.summary.generate-addb-records] ADDB monitors are able to generate summary addb records. 
+
+- [r.addb.monitor.summary.deliver-addb-record-to-stats-service] ADDB monitors send summary addb records to stats service. 
+
+- [r.addb.monitor.summary.deliver-addb-exception-records-to-local-HA] ADDB monitors send exception addb records to local HA (High Availability) component as a fop. 
+
+- [r.addb.monitor.summary.deliver-addb-record-to-addb-service] ADDB monitors send summary addb records to addb service similar to normal addb records. 
+
+- [r.addb.monitor.nesting] There can be ADDB monitors for ADDB records that are generated by some other monitors. 
+
+- [r.addb.monitor.stats-service] Stats service to maintain statistics information received from all the nodes. 
+
+- [r.addb.monitor.stats-service.state] ADDB stats service maintains a state, this state is built from all the addb summary records that this service receives through all the nodes present in the system (cluster). This state basically comprises of all the statistics summary information sent from nodes. 
+
+- [r.addb.monitor.stats-service.query] Client (for eg. m0stats, m0t1fs, etc.) can query to this stats service for getting state information. 
+
+- [r.addb.monitor.stats-service.single-instance] There would be only one instance of the stats service in the cluster/system.
+
+**************************
+ Functional Specification
+**************************
+
+There are two APIs that are globally visible, they are to add/delete monitors.
+
+::
```

Comment:
Please try to use:

>  ```C
> xxx code here
> ```
to insert some code

Reply:
Incorporated.